subprojects/core/src/integTest/groovy/org/gradle/internal/scan/config/BuildScanAutoApplyIntegrationTest.groovy

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.scan.config
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.internal.scan.config.fixtures.BuildScanAutoApplyFixture
+import org.gradle.util.VersionNumber
+import spock.lang.Unroll
+
+import static org.gradle.initialization.StartParameterBuildOptionFactory.BuildScanOption
+import static org.gradle.internal.scan.config.BuildScanPluginAutoApply.BUILD_SCAN_PLUGIN_AUTO_APPLY_VERSION
+import static org.gradle.internal.scan.config.fixtures.BuildScanAutoApplyFixture.BUILD_SCAN_PLUGIN_ID
+import static org.gradle.internal.scan.config.fixtures.BuildScanAutoApplyFixture.PUBLISHING_BUILD_SCAN_MESSAGE_PREFIX
+
+class BuildScanAutoApplyIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
❌ We are missing coverage for when the scan plugin was already applied by an init script. In that case we shouldn't try to apply it again.

Reply:
done